zerver/webhooks/gitlab/view.py

```diff
@@ -265,12 +265,29 @@ def get_pipeline_event_body(payload: Dict[str, Any]) -> str:
             project_homepage,
             build.get('id')
         )
-        builds_status += "* [{}]({}) - {}\n".format(
+        artifact_filename = build.get("artifacts_file").get("filename")
+        if artifact_filename:
+            artifact_download_url = '{}/artifacts/download'.format(build_url)
+            artifact_browse_url = '{}/artifacts/browse'.format(build_url)
+            artifact_string = '  * built artifact: *{}* [[Browse]({})|[Download]({})]\n'.format(
+                artifact_filename,
+                artifact_browse_url,
+                artifact_download_url
+            )
+        else:
+            artifact_string = ''
+        builds_status += "* [{}]({}) - {}\n{}".format(
             build.get('name'),
             build_url,
-            build.get('status')
+            build.get('status'),
+            artifact_string
         )
-    return "[Pipeline]({}) {} with build(s):\n{}.".format(pipeline_url, action, builds_status[:-1])
+    return "[Pipeline ({})]({}) {} with build(s):\n{}.".format(
```

Comment:
It seems like the text that would be hyperlinked here was changed. But... I think that it actually looks better this way, and it _does_ serve a functional purpose, so let's keep it.

Since this change is unrelated to the purpose that this commit is trying to achieve, this change should be split out into a different commit.

Reply:
@Hypro999 - thank you for your feedback! Much appreciated. The proposed change that the `Pipeline` now is a clickable link was done in an earlier, already merged (however not live yet) [PR ](https://github.com/zulip/zulip/pull/14874) from me.
I discussed with colleagues and we are absolutely in favour of seeing the pipeline id as a reference in a gitlab pipline messages too.
We also looked at both possible ways:
[Pipeline (0123456)](some.pipeline.com) vs [Pipeline](some.pipeline.c!m) (0123456) and we all agreed that for us the visual appearance of having the pipeline id in the pipeline link is nicer to look at.
However this is discussable and right now I'm not entirely sure what you meant 😓 

> It seems like the text that would be hyperlinked here was changed. But... I think that it actually looks better this way, and it does serve a functional purpose, so let's keep it.

Are you in favour of getting the id out of the link or are you just saying that it should happen in a commit of its own? Splitting up the commits I understood and will do.

Reply:
> Pipeline (0123456) vs Pipeline (0123456) and we all agreed that for us the visual appearance of having the pipeline id in the pipeline link is nicer to look at.

Yup, I agree 100% (and that's what I was saying in the above message).

> Are you in favour of getting the id out of the link or are you just saying that it should happen in a commit of its own?

The latter. I meant that this change of including the id in the already clickable link is good; but should happen in a different commit.

> Splitting up the commits I understood and will do.

Awesome!

Reply:
My inital commit should be split into two commits of their own purposes now.

Reply:
Yup that looks good now! We can resolve this thread too.